src/core/Transport.js

```diff
@@ -1,21 +1,170 @@
+import {
+  TARGETED_DATA, TARGET_SEARCH_MODES, TARGETED_DATA_URL_BASED, TARGETED_DATA_HTML_BASED
+} from './constants.js';
+
 export default class Transport {
-  constructor(targetUrl) {
-    this.targetUrl = targetUrl;
+
+  static defaultURLHost = '';
+
+  static TARGETED_DATA_RANGE = Object.values(TARGETED_DATA);
+
+  static TARGETED_DATA_URL_BASED_RANGE = Object.values(TARGETED_DATA_URL_BASED);
+
+  static TARGETED_DATA_HTML_BASED_RANGE = Object.values(TARGETED_DATA_HTML_BASED);
+
+  static TARGETED_SEARCH_MODES_RANGE = Object.values(TARGET_SEARCH_MODES);
+
+  /**
+   * @param {String|RCITenancy<Array>} input
+   */
+  constructor(input) {
+    this.targetUrls = [];
+    this.requests = [];
+    if (typeof input === 'string' || input instanceof String) {
+      this.targetUrls.push(input);
+    } else if (Array.isArray(input) && input.length > 0) {
+      this.targetUrls = this.getTargetURLs(input);
+    }
   }
 
-  async execute(event) {
+  static isTargetMatching(target) {
+    if (!Transport.isTargetValid(target)) {
+      return false;
+    }
+
+    // This is an array for cases when multiple targeted data are being found. Ex: canonicalLinks
+    const input = [];
```

Comment:
@robertdumitrescu INFO - I'll assume this is for compatibility with more complex types (future) as all implementations only push one element?